scripts/installer/utils/extractors.py

```diff
@@ -1,161 +1,52 @@
-                extract_ds_object = False
-                current_key = None
-            elif (start_handling_ds_options) and (line == "}"):
-                indices_to_pop.append(i)
-                break
-
-        purge_indices_in_reverse(indices_to_pop)
-        data["data_studio_options"] = data_studio_options
-
-        # 4) Handle multiline arrays. Find opening line with '=' and ending in '['
-        target_index = None
-        indices_to_pop = []
-        for i, line in enumerate(lines_array):
-            if ("=" in line) and (line.strip()[-1] == "["):
-                target_index = i
-                continue
-
-            if target_index is not None:
-                lines_array[target_index] += line.strip()
-                indices_to_pop.append(i)
-
-            if line.strip()[-1] == "]":
-                target_index = None
-
-        purge_indices_in_reverse(indices_to_pop)
-
-        # 5) Convert items to proper python types.
-        for line in lines_array:
-            if "=" in line:
-                key, value = [x.strip() for x in line.split("=", 1)]
-                if value.lower() == "true":
-                    data[key] = True
-                elif value.lower() == "false":
-                    data[key] = False
-                else:
-                    data[key] = ast.literal_eval(value)
-
-    return data
-
-    # # with open('output.json', 'w') as file:
-    # #     json.dump(data, file, indent=4)
-
-
 def get_tfvars_as_json():
-    return convert_tfvars_to_dictionary("terraform.tfvars")
+    """Uses the `tmccombs/hcl2json` Docker image to convert `terraform.tfvars` into JSON format and return it as a Python dictionary."""
+    
+    tfvars_path = os.path.abspath("terraform.tfvars")
+    timestamp = datetime.now().strftime("%Y_%b%d_%I-%M%p")
+    
+    if not os.path.exists(tfvars_path):
+        raise FileNotFoundError("terraform.tfvars file not found in current directory.")
+
+    try:
+        # Create a temporary file for Docker output
+        # delete=True would delete the tmp file as soon as the `with` block exits but we need the file further down
+        with tempfile.NamedTemporaryFile(delete=False, prefix=f"terraform_tfvars_{timestamp}_", suffix=".json", dir="/tmp") as temp_output:
+            output_path = temp_output.name
+
+        cmd = [
+                "docker", "run",
+                "--platform", "linux/amd64",
+                "-i", "--rm",
+                "-v", f"{os.getcwd()}:/tmp:ro",            # read-only mount
+                "--user", "1000:1000",                     # non-root user (adjust UID/GID if needed or user current user)
+                "--network", "none",                       # disable network
+                "tmccombs/hcl2json@sha256:312ac54d3418b87b2ad64f82027483cb8b7750db6458b7b9ebe42ec278696e96",
+                "/tmp/terraform.tfvars"
+            ]
+
+        # Redirect output to temporary JSON file
```

Comment:
I know I asked for the file to be emitted to disk every time but - now that I think about it - I don't see much day-to-day benefit beyond troubleshooting (_especially given the clutter in `/tmp` given the as-is multiple calls_).

I wonder if we should lock this behind a configuration value? 
- Originally I was thinking we use some environment variable, but I'm not thrilled with this as we haven't introduced that mechanism yet and I'm not super keen to make the change.
- Alternatively, we could let the `scripts/installer/utils/logger` level setting control this for us (e.g. "If logger level is debug, emit file to disk."

Just thinking outloud, leaving it to you to decide if/how you want to modify.

Reply:
Updated in latest commit.

Written to file only if `logger` setting is DEBUG.